### ISSUE TITLE
Optimize GeoIP DB download

### ISSUE THREAD

**Issue by prabhatsharma** [openobserve/openobserve#5134]
### Which OpenObserve functionalities are relevant/related to the feature request?

other

### Description

Maxmind GeoIP DB is downloaded at the first install of OpenObserve. It is also downloaded every 2 weeks, whether the DB is being used by any pipeline or not. This has two implications if there are no pipelines using GeoIP DB:

1. Instances of O2 are needlessly updating DB every 2 weeks using download bandwidth ~67 MB of data. 
2. Some instances of O2 may be in a DMZ with no access to the internet, causing failed requests for no reason.
3. Put load on CDN for downloading high amounts of data by thousands of O2 instances.

### Proposed solution

1. Download GeoIP DB on the first install
2. Biweekly downloads should happen only if any pipeline is using GeoIP DB

### Alternatives considered

Do nothing

**Comment by gaby** [comment]
While fixing this can an ENV be added to disable the download for DMZ deployments?

**Comment by taimingl** [comment]
> While fixing this can an ENV be added to disable the download for DMZ deployments?

Yes. You can set "ZO_MMDB_DISABLE_DOWNLOAD" to be true to disable it.
